### DISCUSSION TITLE
Need help，how to trigger 'Data Provider' update after Table Row edit

### DISCUSSION THREAD

**Discussion by yoyogis** [Budibase/budibase#14600] [Help] [answered]
### Discussed in https://github.com/Budibase/budibase/discussions/14584

<div type='discussions-op-text'>

<sup>Originally posted by **yoyogis** September 15, 2024</sup>
I created a screen with a table and a pie chart. Their data points to the same DB table. After I change the value in a table cell, how can I trigger the "Data Provider" to update and refresh the pie chart?
<img width="1152" alt="image" src="https://github.com/user-attachments/assets/2985a564-a883-412b-85f4-28c235aa0e79">
</div>

**Comment by linear** [comment]
<p><a href="https://linear.app/budibase/issue/BUDI-8656/need-help，how-to-trigger-data-provider-update-after-table-row-edit">BUDI-8656 Need help，how to trigger 'Data Provider' update after Table Row edit</a></p>

**Comment by mikesealey** [comment, marked as answer]
Hey @yoyogis I've migrated this to a Help Discussion instead of a bug report.

As for your issue; inline editing wont trigger a refresh of the data as you're hoping. Instead, you should have an on-row-click action to open a modal or side panel that updates the row using a form. Add a "refresh data provider" action to the save button in the form, triggering a refresh of the Data Provider that the Pie Chart is getting it's data from.

If you're using a Chart Block you may need to eject the block to point it at a different data provider.

https://docs.budibase.com/docs/what-is-budibase
